test/unit/math/mix/scal/fun/abs_test.cpp

```diff
@@ -1,101 +1,18 @@
-#include <stan/math/mix/scal.hpp>
-#include <gtest/gtest.h>
-#include <boost/math/special_functions/fpclassify.hpp>
-#include <test/unit/math/rev/scal/fun/util.hpp>
-#include <test/unit/math/mix/scal/fun/nan_util.hpp>
-#include <limits>
-
-TEST(AgradFwdAbs, FvarVar_1stDeriv) {
-  using stan::math::fvar;
-  using stan::math::var;
-  using std::abs;
-
-  fvar<var> x(2.0, 1.0);
-  fvar<var> a = abs(x);
-
-  EXPECT_FLOAT_EQ(2.0, a.val_.val());
-  EXPECT_FLOAT_EQ(1.0, a.d_.val());
-
-  AVEC y = createAVEC(x.val_);
-  VEC g;
-  a.val_.grad(y, g);
-  EXPECT_FLOAT_EQ(1.0, g[0]);
-}
-TEST(AgradFwdAbs, FvarVar_2ndDeriv) {
-  using stan::math::fvar;
-  using stan::math::var;
-  using std::abs;
-
-  fvar<var> x(2.0, 1.0);
-  fvar<var> a = abs(x);
-
-  AVEC z = createAVEC(x.val_);
-  VEC h;
-  a.d_.grad(z, h);
-  EXPECT_FLOAT_EQ(0.0, h[0]);
-}
-
-TEST(AgradFwdAbs, FvarFvarVar_1stDeriv) {
-  using stan::math::fvar;
-  using stan::math::var;
-  using std::abs;
-
-  fvar<fvar<var> > y;
-  y.val_ = fvar<var>(4.0, 1.0);
-
-  fvar<fvar<var> > b = abs(y);
-
-  EXPECT_FLOAT_EQ(4.0, b.val_.val_.val());
-  EXPECT_FLOAT_EQ(1.0, b.val_.d_.val());
-  EXPECT_FLOAT_EQ(0.0, b.d_.val_.val());
-  EXPECT_FLOAT_EQ(0.0, b.d_.d_.val());
-
-  AVEC z = createAVEC(y.val_.val_);
-  VEC h;
-  b.val_.val_.grad(z, h);
-  EXPECT_FLOAT_EQ(1.0, h[0]);
-}
-TEST(AgradFwdAbs, FvarFvarVar_2ndDeriv) {
-  using stan::math::fvar;
-  using stan::math::var;
-  using std::abs;
-
-  fvar<fvar<var> > y;
-  y.val_ = fvar<var>(4.0, 1.0);
-
-  fvar<fvar<var> > b = abs(y);
-
-  AVEC z = createAVEC(y.val_.val_);
-  VEC h;
-  b.val_.d_.grad(z, h);
-  EXPECT_FLOAT_EQ(0.0, h[0]);
-}
-TEST(AgradFwdAbs, FvarFvarVar_3rdDeriv) {
-  using stan::math::fvar;
-  using stan::math::var;
-  using std::abs;
-
-  fvar<fvar<var> > y;
-  y.val_.val_ = 4.0;
-  y.val_.d_ = 1.0;
-  y.d_.val_ = 1.0;
-
-  fvar<fvar<var> > b = abs(y);
-
-  AVEC z = createAVEC(y.val_.val_);
-  VEC h;
-  b.d_.d_.grad(z, h);
-  EXPECT_FLOAT_EQ(0.0, h[0]);
-}
-
-struct abs_fun {
-  template <typename T0>
-  inline T0 operator()(const T0& arg1) const {
-    return abs(arg1);
-  }
-};
-
-TEST(AgradFwdAbs, abs_NaN) {
-  abs_fun abs_;
-  test_nan_mix(abs_, false);
+#include <test/unit/math/test_ad.hpp>
+
+TEST(mixScalFun, abs) {
+  auto f = [](const auto& x) { return stan::math::abs(x); };
+  stan::test::expect_common_nonzero_unary(f);
+  stan::test::expect_value(f, 0);
+  stan::test::expect_value(f, 0.0);
+
+  stan::test::expect_ad(f, -3);
```

Comment:
Isn't this function vectorized? Was there a special mechanism for testing vectorized functions?

Reply:
I super-apologize cause I thought you responded to this and now I'm not finding the response. I vaguely remember you saying this is covered (presumably there's extensive testing of the mechanism that does the vectorization). Was that is?

Reply:
No need to apologize at all!  No, I forgot to respond.  `abs` is not vectorized.  It was deprecated in favor of `fabs`.  We may want to change that now that we're better at type infererence in C++---it used to be a struggle just to get things to compile with these functions that were in some top-level namespaces and not others and had return-type dependendent signatures (int returns int, double returns double).

And no rush on merging this---it's just testing.